### DISCUSSION TITLE
POST to endpoint returning 201 but no new DB row?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#9003] [Questions] [answered]
Hi all

I'm just getting to grips with Supabase, loving it so far. I've got a test table `lists` which a couple of columns, `id` & `name`. My understanding is that this should create the appropriate CRUD endpoints. When I make a GET request on `/rest/v1/lists` I do indeed get the rows I expect. However, submitting a POST with a valid body (excluding ID as it looks as though this is autoincrementing) to the same endpoint returns a 201, but a new row is not created in my DB.

Does anyone have any idea what I'm doing wrong? If I wasn't getting a 201 I'd think something is amiss, but apart from no new row being created, all seems fine! Any help that anyone can offer would be great, thanks!

**Comment by steve-chavez** [comment]
Hello,

Can you share how are you making the request? (curl snippet, etc)

Also try using the `Prefer: return=representation` header so you can see the inserted row.

**Comment by GaryAustin1** [comment]
@steve-chavez 
We solved this on Discord....

If you have a bad body format, extra escapes in this case, PostgREST seems to ignore it and inserts nothing and returns 201 (OK) because it needed to insert nothing.  Not sure on the actual logistics behind no insert and still getting a 201.

**Comment by ghost** [comment, marked as answer]
Sorry, I meant to close this after speaking to @GaryAustin1 on Discord! Yes, receiving a 201 Created when nothing has in fact been created does indeed sound counterintuitive. Looks like it's there are a couple of longstanding issues surrounding it though [https://github.com/PostgREST/postgrest/issues/1323](https://github.com/PostgREST/postgrest/issues/1323) & [https://github.com/PostgREST/postgrest/issues/1070](https://github.com/PostgREST/postgrest/issues/1070)

**Reply by steve-chavez** [reply]
Ah, thanks! We'll fix in https://github.com/PostgREST/postgrest/issues/1323.
